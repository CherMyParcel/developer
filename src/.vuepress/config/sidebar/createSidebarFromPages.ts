import {Language} from '../../plugins/parseTranslations/shared';
import {MyPaSidebarConfig} from '@mptheme/config.types';
import {Page} from 'vuepress';
import {addPageToSidebar} from './addPageToSidebar';

export const createSidebarFromPages = (
  pages: Page[],
  languages: Language[],
  defaultLanguage: Language,
): MyPaSidebarConfig => {
  const sidebar: MyPaSidebarConfig = {};

  pages
    .sort((pageA, pageB) => pageA.path.localeCompare(pageB.path))
    .forEach((page) => {
      addPageToSidebar(page, languages, defaultLanguage, sidebar);
    });

  return sidebar;
};
