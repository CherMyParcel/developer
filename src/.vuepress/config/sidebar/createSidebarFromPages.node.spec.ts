import { describe, expect } from 'vitest';
import { Page } from 'vuepress';
import { createSidebarFromPages } from './createSidebarFromPages';

const languages = [
  {
    alpha2: 'gb',
    localeShort: 'en',
    locale: 'en-GB',
    human: 'English',
    path: '/',
  },
  {
    alpha2: 'nl',
    localeShort: 'nl',
    locale: 'nl-NL',
    human: 'Nederlands',
    path: '/nl/',
  },
];

// Note: don't fix the order. This is on purpose and reflects the actual behavior of vuepress.
const pages: Partial<Page>[] = [
  { path: '/about.html', title: 'About Us' },
  { path: '/contact.html', title: 'Contact' },
  { path: '/integrations.html', title: 'Integrations' },
  { path: '/', title: 'Go Beyond Delivery with MyParcel.' },
  { path: '/api-reference/01.requests.html', title: '1. Requests' },
  { path: '/api-reference/02.responses.html', title: '2. Responses' },
  { path: '/api-reference/', title: 'API Reference' },
  { path: '/documentation/10.woocommerce.html', title: 'WooCommerce' },
  { path: '/documentation/50.php-sdk.html', title: 'PHP SDK' },
  { path: '/documentation/51.javascript-sdk.html', title: 'JavaScript SDK' },
  { path: '/documentation/60.delivery-options.html', title: 'Delivery Options' },
  { path: '/documentation/', title: 'Documentation' },
  { path: '/nl/integraties.html', title: 'Integraties' },
  { path: '/nl/', title: 'Ga Beyond Delivery met MyParcel.' },
  { path: '/documentation/52.pdk/01.backend/01.php.html', title: 'PHP' },
  { path: '/documentation/52.pdk/01.backend.html', title: 'PDK backend' },
  { path: '/documentation/52.pdk/02.frontend.html', title: 'PDK Frontend' },
  { path: '/documentation/52.pdk/', title: 'Plugin Development Kit (PDK)' },
  { path: '/nl/documentatie/10.woocommerce.html', title: 'WooCommerce' },
];

describe('createSidebarFromPages', () => {
  it('should create sidebar from pages', () => {
    const sidebar = createSidebarFromPages(pages as Page[], languages, languages[0]);

    expect(sidebar).toMatchSnapshot();
  });
});
