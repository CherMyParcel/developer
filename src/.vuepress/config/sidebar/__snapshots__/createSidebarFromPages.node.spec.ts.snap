// Vitest Snapshot v1

exports[`createSidebarFromPages > should create sidebar from pages 1`] = `
{
  "/": {
    "/about": {
      "collapsible": false,
      "link": "/about.html",
      "text": "About Us",
    },
    "/api-reference": {
      "children": [
        {
          "collapsible": false,
          "link": "/api-reference/01.requests.html",
          "text": "1. Requests",
        },
        {
          "collapsible": false,
          "link": "/api-reference/02.responses.html",
          "text": "2. Responses",
        },
      ],
      "collapsible": false,
      "link": "/api-reference/",
      "text": "API Reference",
    },
    "/contact": {
      "collapsible": false,
      "link": "/contact.html",
      "text": "Contact",
    },
    "/documentation": {
      "children": [
        {
          "collapsible": false,
          "link": "/documentation/10.woocommerce.html",
          "text": "WooCommerce",
        },
        {
          "collapsible": false,
          "link": "/documentation/50.php-sdk.html",
          "text": "PHP SDK",
        },
        {
          "collapsible": false,
          "link": "/documentation/51.javascript-sdk.html",
          "text": "JavaScript SDK",
        },
        {
          "children": [
            {
              "children": [
                {
                  "collapsible": true,
                  "link": "/documentation/52.pdk/01.backend/01.php.html",
                  "text": "PHP",
                },
              ],
              "collapsible": true,
              "link": "/documentation/52.pdk/01.backend.html",
              "text": "PDK backend",
            },
            {
              "collapsible": true,
              "link": "/documentation/52.pdk/02.frontend.html",
              "text": "PDK Frontend",
            },
          ],
          "collapsible": false,
          "link": "/documentation/52.pdk/",
          "text": "Plugin Development Kit (PDK)",
        },
        {
          "collapsible": false,
          "link": "/documentation/60.delivery-options.html",
          "text": "Delivery Options",
        },
      ],
      "collapsible": false,
      "link": "/documentation/",
      "text": "Documentation",
    },
    "/integrations": {
      "collapsible": false,
      "link": "/integrations.html",
      "text": "Integrations",
    },
    "collapsible": false,
    "link": "/",
    "text": "Go Beyond Delivery with MyParcel.",
  },
  "/nl": {
    "/documentatie": {
      "children": [
        {
          "collapsible": false,
          "link": "/nl/documentatie/10.woocommerce.html",
          "text": "WooCommerce",
        },
      ],
    },
    "/integraties": {
      "collapsible": false,
      "link": "/nl/integraties.html",
      "text": "Integraties",
    },
    "collapsible": false,
    "link": "/nl/",
    "text": "Ga Beyond Delivery met MyParcel.",
  },
}
`;
